### ISSUE TITLE
Loadworld error, can't enter the game

### ISSUE THREAD

**Issue by congiola88** [SpikeHimself/XPortal#72]
**Describe the bug**
Write a clear and concise description of what's going wrong.

**To Reproduce**
Explain the steps needed to reproduce the behaviour.

**Expected behaviour**
Describe what you expected to happen (if it's not obvious).

**LogOutput file(s)**
Please have `Debug` output enabled. Attach the file called `LogOutput.log` which can be found in the directory `Valheim\BepInEx\`.
If you play on a server, include the server's `LogOutput.log` as well.

**Version**
Which version of XPortal are you experiencing your problem with?

**Environment**
Choose from the following options where you are experiencing the problem (leave only the ones you know about):
* Dedicated server
* Multiplayer Host (my game is the server)
* Multiplayer Client (I joined another server/host)
* Singleplayer

**Other mods**
Do you experience the issue also when you do not have any other mods installed? If applicable, please provide the full list of mods that you are using.

**Priority**
Estimate how important this problem is. Choose Medium if you don't know.
* Critical (my world is suffering permanent damage from XPortal)
* High (XPortal does not work at all anymore)
* Medium (XPortal kind of still works, but it's not pleasant to use)
* Low (XPortal is still working fine, but sometimes a thing goes wrong)

**Additional information**
Add any addtional information here. Anything that you think might help is appreciated.

![loi valheim](https://github.com/SpikeHimself/XPortal/assets/147020234/a429c206-d900-4e28-9bec-65ce85186b3c)
[LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12812243/LogOutput.log)

**Comment by congiola88** [comment]
I think you should have your own discord channel to support users when they encounter mod errors or don't understand how to use the mod

**Comment by congiola88** [comment]
I think you should have your own discord channel to support users when they encounter mod errors or don't understand how to use the mod

**Comment by SpikeHimself** [comment]
Hi @congiola88 thank you for reporting the error. Can you please fill in the template? You didn't answer any of the questions, so I don't have any information to go on.

**Comment by congiola88** [comment]
I can't write, I don't know what this page is doing, in general, your portal mod is broken, I have to delete it to get into the game.

**Comment by SpikeHimself** [comment]
I understand that you are getting an error. It is specific to your situation - other players do not get this error. Therefore I need more information from you. Please fill out the template that I provided (just edit your post and answer the questions).

**Comment by congiola88** [comment]
That's it, that's all I'm reporting, anyway I deleted it...hopefully you will have your own discord channel to support everyone more easily, or maybe I've already met on discord :))

**Comment by SpikeHimself** [comment]
I cannot help you with the issue that is happening in your game, without additional information.

Please fill in the questions in the template and attach a log file of the game in which the error happened, with debug logging enabled.

**Comment by ConflyctX** [comment]
I had the same error which prevented the world from loading, and it tried to place location again when starting that save game.
I am using Expand world with a radius of 35km though so it might be a problem with that. Hard to know since the other guy didn't fill in the modding info.

Removing Xportal made the world load in again.

**Comment by SpikeHimself** [comment]
Hi @ConflyctX, thank you for adding that. Are you able to share your logs with debug logging enabled, so that I can have a look?

To enable debug logging, see [here](https://github.com/SpikeHimself/resources/wiki/BepInEx:-Enable-debug-logging).

**Comment by mischab1** [comment]
Error is happening to me too.  It took most of the day but I'm finally able to semi-consistently reproduce the bug.  Here is the info the original posted didn't give you.

Describe the bug:
XPortal causing Exception while loading world which prevents world from loading

To Reproduce:
Bug is not consistantly reproduceable but it does happen frequently enough that I was able to create 2 examples. (See ex1 & ex2 StepsTaken.txt files) Once a world is broken like this then most of the time it will stay broken ("ex3 LogOutput.log") until you open the world with XPortal disabled. But not always. Twice in my testing, after restarting Valheim with XPortal still enabled, it was able to open the 'broken' world without any errors.

LogOutput:
"ex1 LogOutput.log" and "ex2 LogOutput.log" files are the result of the corresponding ex1 & ex2 StepsTaken.txt files.

XPortal Version: 1.2.17

Environment:
Steam version of Valheim via Thunderstore 1.32.2. worlds are saved to cloud

Other mods:
BepInExPack_Valheim v5.4.2200
Jotunn v2.14.3
Official_BepInEx_ConfigurationManager v18.0.1

Priority:
Medium (XPortal kind of still works, but it's not pleasant to use)

[ex1 LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12839912/ex1.LogOutput.log)
[ex1 StepsTaken.txt](https://github.com/SpikeHimself/XPortal/files/12839913/ex1.StepsTaken.txt)
[ex2 LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12839914/ex2.LogOutput.log)
[ex2 StepsTaken.txt](https://github.com/SpikeHimself/XPortal/files/12839915/ex2.StepsTaken.txt)
[ex3 LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12839916/ex3.LogOutput.log)
![Thunderstore Mod List](https://github.com/SpikeHimself/XPortal/assets/1222506/5476ffeb-cfc5-4db8-b360-81eda517aced)
![Valheim world list](https://github.com/SpikeHimself/XPortal/assets/1222506/c059d07c-4fb3-4819-839b-b2d0dd8d4be6)

Last two attachments are .png files

While testing this I discovered a 2nd bug which I'll open as a different ticket.

**Comment by mischab1** [comment]
Also, I don't remember if this is new behavior as a result of the last Valheim update but but if you link portal 1 to portal 2 and then go thru the portal, the receiving portal 2, even though it does **not** have a destination set, is active. You can turn around and go thru it back to portal 1 without setting a destination.  (If you want me to open an issue for this, let me know.)

**Comment by SpikeHimself** [comment]
Thank you for adding that @mischab1, I'll look into this ASAP.

**Comment by SpikeHimself** [comment]
Hi @mischab1, could you please test with this debug release (attached)?

Please also update Jotunn and BepInEx to the latest versions if you hadn't already.

[XPortal-debug-v1.2.18-issue72-01.zip](https://github.com/SpikeHimself/XPortal/files/12848686/XPortal-debug-v1.2.18-issue72-01.zip)

This update addresses both issues you mentioned:
- A portal that you did not configure will no longer connect to its counterpart automagically
- The game will not crash when loading a world when XPortal can't find a portal it's trying to find

**Comment by mischab1** [comment]
I've done a bunch of testing and I can't re-create either issue.  However, 9 times out of 10, if I start with no portals connected to any other portal and then daisy-chain them, 1 -> 2 -> 3 and then logout; open a different world; don't do anything; log out of that world and back into my test world, it's forgotten the last link in the daisy chain.  

I've tested this with 3 portals and with 4.  Link 1 to 2 and then go thru the portal; link 2 to 3 and go thru the portal; link 3 to 4 and go thru the portal. Log out, open different world, log out, re-open this world. Run back over to where portal 3 is and it's no longer linked to portal 4.  If portal 4 was already linked to another portal when portal 3 was linked to it, this doesn't happen.

**Comment by SpikeHimself** [comment]
Good to hear that the issue is solved, I'll release that version later today.

> if I start with no portals connected to any other portal and then daisy-chain them, 1 -> 2 -> 3 and then logout; open a different world; don't do anything; log out of that world and back into my test world, it's forgotten the last link in the daisy chain

That's an interesting observation, can you provide a log file of such a test run?

**Comment by mischab1** [comment]
[LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12858752/LogOutput.log)

Above is my last log from yesterday.  I was testing a bunch of different ways to link portals some that worked and some that didn't.  I'll try to do a new test just showing the issue this evening after work.

**Comment by mischab1** [comment]
Here is a shorter log with 3 tests.  My test world has 4 portals named p1 thru p4. I reset all of them to none before starting this test.

Start modded valheim with:
- BenInExPack_Valheim v5.4.2200
- Jotunn v2.14.4
- Official_BepInEx_ConfigurationManager v18.0.1
- XPortal debug v1.2.18)

**Test 1**
open world TestXPortal
connect portal p1 to portal p2, go thru portal
connect portal p2 to portal p3, go thru portal
logout of world
open a different world and close it
reopen world TestXPortal
p2 lost connection to p3

**Test 2**
walk back to portal p2 and re-connect it to p3, go thru portal
logout of world
open a different world and close it
reopen world TestXPortal
p2 remembers connection to p3

**Test3**
connect portal p3 to p4, go thru portal
(The extra portal triggers are because I stepped thru p4 twice killing a Neck)
logout of world
open a different world and close it
reopen world TestXPortal
p3 forgot p4
quit valheim

[LogOutput.log](https://github.com/SpikeHimself/XPortal/files/12865312/LogOutput.log)

**Comment by SpikeHimself** [comment]
Thank you very much for the extended testing, I will look into this soon. Life is lifing again and I'm a little short on time.

**Comment by SpikeHimself** [comment]
Actually @mischab1 would you mind creating a new issue regarding the portal chaining issue you posted?

I will close this one regarding the world load error, as I have just released v1.2.18 which addresses that issue.
